packages/modules/viewer-react/src/types.ts

```diff
@@ -246,3 +255,20 @@ export type ViewerDefaultFrontstageConfig = Pick<
   StandardFrontstageProps,
   "hideNavigationAid" | "hideStatusBar" | "hideToolSettings"
 >;
+/**
+ * Custom backend and rpc configuration
+ */
+export type BackendConfiguration = {
+  defaultBackend?: RequireOneOf<DefaultBackend, "rpcInterfaces" | "config">;
+  customBackends?: CustomBackend[];
+};
+
+export type DefaultBackend = {
+  rpcInterfaces?: RpcInterfaceDefinition<RpcInterface>[]; // will be combined with a set of default interfaces
+  config?: Partial<BentleyCloudRpcParams>;
+};
+
+export type CustomBackend = {
+  rpcInterfaces: RpcInterfaceDefinition<RpcInterface>[]; // will be combined with a set of default interfaces
```

Comment:
You're thinking either/or? If custom backend is provided - those are the only rpc interfaces which get registered?

Reply:
we're kinda just guessing i guess since this is all hypothetical atm...
@calebmshafer does it makes sense to register the defaul web rpcs across every backend?

Reply:
> does it makes sense to register the defaul web rpcs across every backend?

Nope, I think if you're registering a custom backend you should define all of the rpcs you want to use. The only reason it makes sense to register by default now is with the Visualization service as the default we register all the ones it supports:

- ReadRpcInterface
- TileRpcInterface
- PresentationRpcInterface
- ECSchemaRpcInterface

Reply:
the viewer has never registered the `ECSchemaRpcInterface` by default. is that something we need to add?

Reply:
@calebmshafer